libswc/seat-ws.c

```diff
@@ -0,0 +1,460 @@
+/* swc: libswc/seat-ws.c
```

Comment:
Since this shared some of the same code as the original `seat.c`, do you think it's possible to split this file up in some way to avoid the duplication?

Reply:
I can do that (it seems the shared code mainly makes use of seat_base anyway), but it probably makes sense to do it on another branch?

Reply:
Whatever you prefer. I'm fine to merge this as-is and leave factoring out the shared code for a separate PR.

Reply:
Sure! Thanks.